File: static/civics/angular/services/Categories.js
angular.module('civics.categories_service', [])

.factory('Categories', function(){

    this.topics = {
      'dc': 'Desarrollo comunitario',
      'au': 'Arte urbano',
      'cl': 'Cultura libre',
      'ds': 'Deporte / Salud / Cuidados',
      'id': 'Igualdad / Derechos / Memoria',
      'ec': 'Ecología / Consumo',
      'oe': 'Otras economías',
      'ee': 'Educación expandida',
      'ct': 'Ciencia / Tecnología',
      'ms': 'Movilidad sostenible',
      'pg': 'Política y gobernanza',
      'up': 'Urbanismo / Patrimonio',
      'pc': 'Periodismo comunitario',
      'in': 'Infancia',
      'di': 'Digitalización / Datos Abiertos',
      'fe': 'Feminismos',
    };

    this.spaces = {
      'cc': 'Centro cultural / comunitario',
      'ei': 'Efímero e itinerante',
      'it': 'Intercambio / Trueque',
      'di': 'Digital',
      'ea': 'Encuentros / Acciones',
      'ep': 'Escuelas populares',
      'hu': 'Huerto urbano / rural',
      'iu': 'Intervenciones urbanas',
      'mc': 'Medios de comunicación comunitaria',
      'ms': 'Mercado social / Comercios',
      'se': 'Solares / Espacios recuperados',
      'em': 'Labs / Colaborativos / Maker',
    };

    this.agents = {
      'im': 'Iniciativas municipales / Gobierno',
      'uo': 'Universidades / ONG / Fundaciones',
      'oi': 'Organismos internacionales',
      'es': 'Empresa social / Startup',
      'ic': 'Iniciativa ciudadana',
      'ja': 'Juntas / Asociaciones de vecinos',
      'cc': 'Iniciativas ciudadanas inactivas'
    };

    this.activities = {
      'au': 'Audiovisual',
      'cu': 'Curso / Convocatoria',
      'di': 'Digital',
      'en': 'Encuentro',
      'ev': 'Evento',
      'ex': 'Exposicion',
      'fi': 'Fiesta / Concierto',
      'pu': 'Publicación',
      'ta': 'Taller',
    };

    this.ods = {
      '1'  : 'Fin de la pobreza',
      '2'  : 'Hambre cero',
      '3'  : 'Salud y bienestar',
      '4'  : 'Educación de calidad',
      '5'  : 'Igualdad de género',
      '6'  : 'Agua limpia y saneamiento',
      '7'  : 'Energía asequible y no contaminante',
      '8'  : 'Trabajo decente y crecimiento económico',
      '9'  : 'Industria, innovación e infraestructura',
      '10' : 'Reducción de las desigualdades',
      '11' : 'Ciudades y comunidades sostenibles',
      '12' : 'Producción y consumo responsables',
      '13' : 'Acción por el clima',
      '14' : 'Vida submarina',
      '15' : 'Vida de ecosistemas terrestres',
      '16' : 'Paz, justicia e instituciones sólidas',
      '17' : 'Alianzas para lograr los objetivos',
    };

    this.city_initiatives = {};
    this.city_events = {};
    this.cities = {};

    this.addInitiativeCity = function(country, name, id, coords){
        if(!this.city_initiatives[country])
            this.city_initiatives[country] = {}
        if(!this.city_initiatives[country][id]) {
            this.city_initiatives[country][id] = {
                'name'   : name,
                'coords' : [ coords[1], coords[0] ],
            };
            this.cities[id] = name;
        }
    }

    this.addEventCity = function(country, name, id, coords){
        if(!this.city_events[country])
            this.city_events[country] = {}
        if(!this.city_events[country][id]) {
            this.city_events[country][id] = {
                'name'   : name,
                'coords' : [ coords[1], coords[0] ],
            };
            this.cities[id] = name;
        }
    }

    return this;

})

File: apps/models/categories.py
from django.utils.translation import ugettext_lazy as _

"""
The different categories used by the Models of Civics
"""

TOPICS = (
    ('CO', _('COVID-19')),
    ('DC', _('Desarrollo comunitario')),
    ('AU', _('Arte urbano')),
    ('CL', _('Cultura libre')),
    ('DS', _('Deporte / Salud / Cuidados')),
    ('ID', _('Igualdad / Derechos / Memoria')),
    ('EC', _('Ecología / Consumo')),
    ('OE', _('Otras economías')),
    ('EE', _('Educación expandida')),
    ('CT', _('Ciencia / Tecnología')),
    ('MS', _('Movilidad sostenible')),
    ('PG', _('Política y gobernanza')),
    ('UP', _('Urbanismo / Patrimonio')),
    ('PC', _('Periodismo comunitario')),
    ('IN', _('Infancia')),
    ('DI', _('Digitalización / Datos Abiertos')),
    ('FE', _('Feminismos')),
)

SPACES = (
    ('CC', _('Centro cultural / comunitario')),
    ('EI', _('Efímero e itinerante')),
    ('IT', _('Intercambio / Trueque')),
    ('DI', _('Digital')),
    ('EA', _('Encuentros / Acciones')),
    ('EP', _('Escuelas populares')),
    ('HU', _('Huerto urbano/rural')),
    ('IU', _('Intervenciones urbanas')),
    ('MC', _('Medios de comunicación comunitaria')),
    ('MS', _('Mercado social / Comercios')),
    ('SE', _('Solares / Espacios recuperados')),
    ('EM', _('Labs / Colaborativos / Maker')),
)

ACTIVITIES = (
    ('AU', _('Audiovisual')),
    ('CU', _('Curso / Convocatoria')),
    ('DI', _('Digital')),
    ('EN', _('Acción')),
    ('EV', _('Evento')),
    ('EX', _('Exposicion')),
    ('FI', _('Fiesta / Concierto')),
    ('PU', _('Publicación')),
    ('TA', _('Taller')),
)

AGENTS = (
    ('IM', _('Iniciativas municipales / Gobierno')),
    ('UO', _('Universidades / ONG / Fundaciones')),
    ('OI', _('Organismos internacionales')),
    ('ES', _('Empresa social / Startup')),
    ('IC', _('Iniciativa ciudadana')),
    ('JA', _('Juntas / Asociaciones de vecinos')),
    ('CC', _('Iniciativas ciudadanas inactivas'))
)

ODS = (
    ('1', _('Fin de la pobreza')),
    ('2', _('Hambre cero')),
    ('3', _('Salud y bienestar')),
    ('4', _('Educación de calidad')),
    ('5', _('Igualdad de género')),
    ('6', _('Agua limpia y saneamiento')),
    ('7', _('Energía asequible y no contaminante')),
    ('8', _('Trabajo decente y crecimiento económico')),
    ('9', _('Industria, innovación e infraestructura')),
    ('10',_('Reducción de las desigualdades')),
    ('11',_('Ciudades y comunidades sostenibles')),
    ('12',_('Producción y consumo responsables')),
    ('13',_('Acción por el clima')),
    ('14',_('Vida submarina')),
    ('15',_('Vida de ecosistemas terrestres')),
    ('16',_('Paz, justicia e instituciones sólidas')),
    ('17',_('Alianzas para lograr los objetivos')),
)

RESOURCES = (
    ('0', _('Guías')),
    ('1', _('Mapas')),
    ('2', _('Señalética urbana')),
    ('3', _('Forks de civics')),
    ('4', _('Otras plataformas')),
)
